components/server/src/user/user-service.ts

```diff
@@ -144,7 +144,7 @@ export class UserService {
 
             newUser.blocked = !canPass;
         }
-        if (!newUser.blocked && (isFirstUser || this.env.makeNewUsersAdmin)) {
```

Comment:
this is to enable "first-user-is-admin" in self-hosted installation. how would you suggest to proceed with self-hosted?

Reply:
Good question. The removed behavior isn't ideal, because anyone can become admin by being the first to log into a fresh Gitpod deployment (e.g. from a PR).

Instead of always making the first user admin, I'd recommend one of the following options:
- Update your user in the DB
- Temporarily enable "make new users admin" when you install self-hosted
- Implement a "new user admin passlist" array that makes new users admin based on email domain

Reply:
Actually this breaks the installation story of self-hosted installations entirely. There is no need for any UI if the installer need to edit deployments and delete pods, or connecting to the db in background.

 As far as I understand, we want to protect preview environments, right? In that case let's disable setting of the admin flag for the preview environments (aka `kubestage === 'devstaging'`). see https://github.com/gitpod-io/gitpod/blob/c5641184da7f5bb1204b2b038264742fe2f72bad/components/gitpod-protocol/src/env.ts#L18